superset/models/sql_lab.py

```diff
@@ -179,6 +231,53 @@ def raise_for_access(self) -> None:
 
         security_manager.raise_for_access(query=self)
 
+    @property
+    def db_engine_spec(self) -> Type["BaseEngineSpec"]:
+        return self.database.db_engine_spec
+
+    @property
+    def owners_data(self) -> List[Dict[str, Any]]:
+        return []
+
+    @property
+    def uid(self) -> str:
+        return f"{self.id}__{self.type}"
+
+    @property
+    def is_rls_supported(self) -> bool:
+        return False
+
+    @property
+    def cache_timeout(self) -> int:
+        return 0
+
+    @property
+    def column_names(self) -> List[Any]:
+        return [col.get("column_name") for col in self.columns]
+
+    @property
+    def offset(self) -> int:
+        return 0
+
+    @property
+    def main_dttm_col(self) -> Optional[str]:
+        for col in self.columns:
+            if col.get("is_dttm"):
+                return col.get("column_name")  # type: ignore
+        return None
+
+    @property
+    def dttm_cols(self) -> List[Any]:
+        return [col.get("column_name") for col in self.columns if col.get("is_dttm")]
+
+    @property
+    def default_endpoint(self) -> str:
+        return ""
```

Comment:
generate /explore?dataset_id={id}&dataset_type={query}`

Reply:
is this a todo?